src/components/SearchModal/CurrencyList/index.tsx

```diff
@@ -67,7 +68,25 @@ const WarningContainer = styled.div`
 `
 
 function Balance({ balance }: { balance: CurrencyAmount<Currency> }) {
-  return <StyledBalanceText title={balance.toExact()}>{balance.toSignificant(4)}</StyledBalanceText>
+  const { formatNumberOrString } = useFormatter()
+
+  return (
+    <StyledBalanceText title={balance.toExact()}>
+      {formatNumberOrString({
+        input: balance.toExact(),
+        type: [
+          {
+            upperBound: Infinity,
+            formatterOptions: {
+              notation: 'standard',
+              maximumSignificantDigits: 4,
+              minimumSignificantDigits: 1,
+            },
+          },
+        ],
+      })}
```

Comment:
How come this isn't using a pre-defined type?

Reply:
i was just matching the previous implementation